extensions/wikia/CharacterConverter/CharacterConverter.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+class CharacterConverter {
+	/** @var string $dbName */
+	private $dbName;
+
+	/** @var DatabaseBase $readConnection */
+	private $readConnection;
+
+	/** @var DatabaseBase $writeConnection */
+	private $writeConnection;
+
+	/** @var callable $preConversionCallback */
+	private $preConversionCallback;
+
+	private function __construct( string $dbName ) {
+		$this->dbName = $dbName;
+	}
+
+	public static function newFromDatabase( string $dbName ): CharacterConverter {
+		return new self( $dbName );
+	}
+
+	/**
+	 * @param callable $preConversionCallback
+	 */
+	public function registerPreConversionCallback( callable $preConversionCallback ) {
+		$this->preConversionCallback = $preConversionCallback;
+	}
+
+	public function convert( array $columnConfig ) {
+		global $wgUseUnicode;
+
+		// This point forward, use utf8mb4 for all new connections! 🤞
+		$wgUseUnicode = true;
```

Comment:
I hope we will make this variable name more meaningful before the roll-out.
I believe we should store this information somewhere, so we know that this DB is already migrated and should start using UTF db connection. Also, we could put a check that protects us from executing the same DB multiple times.